database/database.go

```diff
@@ -25,17 +25,19 @@ func NewAndMigrate(ctx context.Context, logger log.Logger, config DatabaseConfig
 		logger = log.NewNopLogger()
 	}
 
-	if ctx == nil {
-		ctx = context.Background()
-	}
-
 	// run migrations first
-	if err := RunMigrations(logger, config); err != nil {
+	db, err := RunMigrations(logger, config)
+	if err != nil {
 		return nil, err
 	}
 
+	// In SQLite, we can reuse the same connection
```

Comment:
Can we document this is because `:memory:` creates a different database on each init?